gateway/gateway-requests/src/registration/handshake/gateway.rs

```diff
@@ -22,11 +22,12 @@ impl<'a> GatewayHandshake<'a> {
         ws_stream: &'a mut S,
         identity: &'a crypto::asymmetric::identity::KeyPair,
         received_init_payload: Vec<u8>,
+        validator_urls: Vec<String>,
```

Comment:
I don't like putting validator urls here. I think gateway should obtain the entire verification key on startup. Think about it. If say 10 clients decide to start talking with the gateway, it will have to talk with the same validators all 10 times only to produce exactly the same verification key. So instead I'd say this argument should be just the verification key itself.

Reply:
I see what you mean, but in order to do that optimization, we'll need to have some caching mechanism implemented in the gateway, for refreshing that verification key, right? Until we do that, I'll just make the interface more natural by having the verification key as argument, but the problem of querying the verification key at each handshake will remain.